src/abbfreeathome/freeathome.py

```diff
@@ -186,7 +207,19 @@ async def update_device(self, data: dict):
             except KeyError:
                 continue
 
-    def _get_function_to_device_mapping(self) -> dict[Function, Base]:
+    def _get_function_to_device_mapping(
+        self, virtual_device: bool = False
+    ) -> dict[Function, Base]:
+        if virtual_device:
+            return (
+                FUNCTION_VIRTUAL_DEVICE_MAPPING
+                if not self._device_classes
+                else {
+                    key: value
+                    for key, value in FUNCTION_VIRTUAL_DEVICE_MAPPING.items()
+                    if value in self._device_classes
+                }
+            )
```

Comment:
Small change, this can be simplified by setting a virtual device mapping variable and applying the existing logic.

```
    def _get_function_to_device_mapping(
        self, virtual_device: bool = False
    ) -> dict[Function, Base]:
        _device_mapping = (
            FUNCTION_VIRTUAL_DEVICE_MAPPING
            if virtual_device
            else FUNCTION_DEVICE_MAPPING
        )

        return (
            _device_mapping
            if not self._device_classes
            else {
                key: value
                for key, value in _device_mapping.items()
                if value in self._device_classes
            }
        )
```